evmcore/tx_pool_test.go

```diff
@@ -505,33 +562,281 @@ func TestEIP7702Transactions_InvalidTransactionsReturnAnError(t *testing.T) {
+	db.balances[addrC] = new(uint256.Int).SetUint64(params.Ether)
 
-	return types.NewTx(&types.SetCodeTx{
-		ChainID:   uint256.MustFromBig(chainId),
-		Nonce:     0,
-		GasTipCap: uint256.NewInt(1e10),  // max priority fee per gas
-		GasFeeCap: uint256.NewInt(50e10), // max fee per gas
-		Gas:       250000,                // gas limit for the transaction
-		To:        common.Address{},      // sponsored address
-		Value:     uint256.NewInt(0),     // value transferred in the transaction
-		Data:      nil,                   // No additional data is sent in this transaction
-		AuthList:  authorizations,        // authorizations in the transaction
+	tests := map[string]struct {
+		test    func(*testing.T, *TxPool)
+		pending int
+		queued  int
+	}{
+		"only one transaction from delegating account in flight": {
+			test: func(t *testing.T, pool *TxPool) {
+				db.codeHashes[addrA] = common.BytesToHash([]byte{0xaa})
+
+				// first transaction is accepted
+				if err := pool.addRemoteSync(pricedTransaction(0, 100000, big.NewInt(1), keyA)); err != nil {
+					t.Fatalf("failed to add remote transaction: %v", err)
+				}
+				// second transaction is rejected
+				if err := pool.addRemoteSync(pricedTransaction(1, 100000, big.NewInt(1), keyA)); !errors.Is(err, ErrInflightTxLimitReached) {
+					t.Fatalf("error mismatch: want %v, have %v", ErrInflightTxLimitReached, err)
+				}
+				// gapped transaction is rejected as well
+				if err := pool.addRemoteSync(pricedTransaction(2, 100000, big.NewInt(1), keyA)); !errors.Is(err, ErrInflightTxLimitReached) {
+					t.Fatalf("error mismatch: want %v, have %v", ErrInflightTxLimitReached, err)
+				}
+				// valid replacement succeeds
+				if err := pool.addRemoteSync(pricedTransaction(0, 100000, big.NewInt(10), keyA)); err != nil {
+					t.Fatalf("failed to replace with remote transaction: %v", err)
+				}
+			},
+			pending: 1,
+		},
+		"allow setcode tx with pending authority tx": {
+			test: func(t *testing.T, pool *TxPool) {
+
+				// Send two transactions where the first has no conflicting delegations and
+				// the second should be allowed despite conflicting with the authorities in 1.
+				if err := pool.addRemoteSync(setCodeTx(0, keyA, []unsignedAuth{{1, keyC}})); err != nil {
+					t.Fatalf("failed to add with remote setcode transaction: %v", err)
+				}
+				if err := pool.addRemoteSync(setCodeTx(0, keyB, []unsignedAuth{{1, keyC}})); err != nil {
+					t.Fatalf("failed to add conflicting delegation: %v", err)
+				}
+			},
+			pending: 2,
+		},
+		"allow one tx from pooled delegation": {
+			test: func(t *testing.T, pool *TxPool) {
+				// Verify C cannot originate another transaction when it has a pooled delegation.
+				if err := pool.addRemoteSync(setCodeTx(0, keyA, []unsignedAuth{{0, keyC}})); err != nil {
+					t.Fatalf("failed to add with remote setcode transaction: %v", err)
+				}
+				if err := pool.addRemoteSync(pricedTransaction(0, 100000, big.NewInt(1), keyC)); err != nil {
```

Comment:
The comment above states that this should not be accepted, but the test checks for being accepted. I guess the comment is outdated?